src/deriver/tom/embeddings.py

```
@@ -0,0 +1,237 @@
+    async def save_facts(self, facts: List[str], replace_duplicates: bool = True, similarity_threshold: float = 0.85) -> None:
+        """Save facts to the collection.
+        
+        Args:
+            facts: List of facts to save
+            replace_duplicates: If True, replace old duplicates with new facts. If False, discard new duplicates
+            similarity_threshold: Facts with similarity above this threshold are considered duplicates
+        """
+        for fact in facts:
+            # Create document with duplicate checking
+            try:
+                document = await crud.create_document(
```

Ruff (0.8.2)

Local variable `document` is assigned to but never used

Remove assignment to unused variable `document`

(F841)